src/main/java/org/sonarsource/sonarlint/ls/connected/sync/ServerSynchronizer.java

```diff
@@ -0,0 +1,196 @@
+    this.client = client;
+    this.progressManager = progressManager;
+    this.bindingManager = bindingManager;
+    this.analysisScheduler = analysisScheduler;
+    var syncPeriod = Long.parseLong(StringUtils.defaultIfBlank(System.getenv("SONARLINT_INTERNAL_SYNC_PERIOD"), "3600")) * 1000;
+    this.serverSyncTimer = serverSyncTimer;
+    this.serverSyncTimer.scheduleAtFixedRate(new SyncTask(), syncPeriod, syncPeriod);
+  }
+
+  public void updateAllBindings(CancelChecker cancelToken, @Nullable Either<String, Integer> workDoneToken) {
+    progressManager.doWithProgress("Update bindings", workDoneToken, cancelToken, progress -> {
+      // Clear cached bindings to force rebind during next analysis
+      bindingManager.clearBindingCache();
+      updateBindings(bindingManager.getActiveConnectionsAndProjects(), progress);
+    });
+  }
+
+  private void updateBindings(Map<String, Map<String, Set<String>>> projectKeyByConnectionIdsToUpdate, ProgressFacade progress) {
+    var failedConnectionIds = tryUpdateConnectionsAndBoundProjectStorages(projectKeyByConnectionIdsToUpdate, progress);
+    showOperationResult(failedConnectionIds);
+    triggerAnalysisOfAllOpenFilesInBoundFolders(failedConnectionIds);
+  }
+
+  private void triggerAnalysisOfAllOpenFilesInBoundFolders(Set<String> failedConnectionIds) {
+    bindingManager.forEachBoundFolder((folder, folderSettings) -> {
+      if (!failedConnectionIds.contains(folderSettings.getConnectionId())) {
+        analysisScheduler.analyzeAllOpenFilesInFolder(folder);
+      }
+    });
+  }
+
+  private void showOperationResult(Set<String> failedConnectionIds) {
+    if (failedConnectionIds.isEmpty()) {
+      client.showMessage(new MessageParams(MessageType.Info, "All SonarLint bindings successfully updated"));
+    } else {
+      var connections = String.join(", ", failedConnectionIds);
+      client.showMessage(
+        new MessageParams(MessageType.Error, "Binding update failed for the following connection(s): " + connections + ". Look at the SonarLint output for details."));
+    }
+  }
+
+  private Set<String> tryUpdateConnectionsAndBoundProjectStorages(Map<String, Map<String, Set<String>>> projectKeyByConnectionIdsToUpdate, ProgressFacade progress) {
+    var failedConnectionIds = new LinkedHashSet<String>();
+    projectKeyByConnectionIdsToUpdate.forEach(
+      (connectionId, projectKeys) -> tryUpdateConnectionAndBoundProjectsStorages(projectKeyByConnectionIdsToUpdate, progress, failedConnectionIds, connectionId, projectKeys));
+    return failedConnectionIds;
+  }
+
+  private void tryUpdateConnectionAndBoundProjectsStorages(Map<String, Map<String, Set<String>>> projectKeyByConnectionIdsToUpdate, ProgressFacade progress,
+    Set<String> failedConnectionIds, String connectionId, Map<String, Set<String>> branchNamesByProjectKey) {
+    progress.doInSubProgress(connectionId, 1.0f / projectKeyByConnectionIdsToUpdate.size(), subProgress -> {
```

Comment:
Just to verify, will `projectKeyByConnectionIdsToUpdate.size()` always be >0?

Reply:
Yes. I changed the code a bit to make it more obvious